File: 70. Climbing Stairs/optimised2.ts

function climbStairs(n: number): number {
    let a = 1;
	let b = 1;
	for (let i = 0; i < n - 1; i++) {
		[a, b] = [b, a + b];
	}

	return b;
};

console.log(climbStairs(3));
console.log(climbStairs(4));

File: 70. Climbing Stairs/optimised2.py
#!/usr/bin/env python
def climbStairs(n: int) -> int:
	if n <= 2:
		return n
	
	a = 1
	b = 1

	for _ in range(n - 1):
		a, b = b, a + b
	return b

print(climbStairs(3))